roles/bigip_user/tasks/main.yaml

```diff
@@ -1,22 +1,44 @@
 ---
+# This will fail as we require password attribute
+- name: Create user, no password
+  bigip_user:
+      server: "{{ inventory_hostname }}"
+      server_port: "{{ bigip_port }}"
+      user: "{{ bigip_username }}"
+      password: "{{ bigip_password }}"
+      partition_access:
+        - "Common:guest"
+      validate_certs: "{{ validate_certs }}"
+      username_credential: "{{ username_credential }}"
+  ignore_errors: true
+  register: result
+
+# The default setting for update_password is "on_create", so this
+# check should NOT return a "changed" event
+- name: Assert create user, no password
+  assert:
+      that:
+          - not result|changed
 
 - name: Create user
   bigip_user:
       server: "{{ inventory_hostname }}"
       server_port: "{{ bigip_port }}"
       user: "{{ bigip_username }}"
       password: "{{ bigip_password }}"
+      partition_access:
+        - "Common:guest"
       validate_certs: "{{ validate_certs }}"
       username_credential: "{{ username_credential }}"
       password_credential: "{{ password_credential }}"
-      update_password: "on_create"
   register: result
 
 - name: Assert Create user
   assert:
       that:
           - result|changed
 
+
 - name: Create user - Idempotent check
```

Comment:
the idempotent check needs to reflect, exactly, the options specified in the test that it is checking for itempotency. so this test needs to be identical to the one above it.